### DISCUSSION TITLE
Why this json gets parser as an array instead of an object?

### DISCUSSION THREAD

**Discussion by danielmevi** [nlohmann/json#2568] [Q&A] [answered]
Why this parsing returns an json array? And what can I do without changing the json string to have it as a object?
```
auto expected_json2{nlohmann::json::parse(
      "{\"domainName\":\"Settings\",\"domains\":[],\"settings\":"
      "[{\"name\":\"ambient\",\"value\":50},{\"name\":"
      "\"PositionX\",\"value\":5}]}")};
```

result
```
[
    {
        "domainName": "Settings",
        "domains": [],
        "settings": [
            {
                "name": "ambient",
                "value": 50
            },
            {
                "name": "PositionX",
                "value": 5
            }
        ]
    }
]
```

**Comment by gregmarr** [comment, marked as answer]
Can you try `auto expected_json2 = nlohmann::json::parse(...` instead?

**Reply by danielmevi** [reply]
wow, that worked. 
Whats the nature of that?

**Reply by gregmarr** [reply]
The `{}` around it converts it into an initializer list of type `json`, which is used to initialize an array.

**Reply by danielmevi** [reply]
I couldn't find any documentation about it, therefore that difference is a bit tricky considering the already complex c++ initialization. But t worked, so thanks a lot

**Reply by nlohmann** [reply]
The documentation is the "feature" described in https://github.com/nlohmann/json#json-as-first-class-data-type that you can create the JSON array `[1,2,3]` with

```cpp
json j = {1, 2, 3};
```

**Reply by gregmarr** [reply]
@danielmevi you're welcome.  Glad you got it working.
